Source/JavaScriptCore/heap/MarkedBlockInlines.h

```diff
@@ -291,7 +291,7 @@ void MarkedBlock::Handle::specializedSweep(FreeList* freeList, MarkedBlock::Hand
     auto setBits = [&] (bool isEmpty) ALWAYS_INLINE_LAMBDA {
         Locker locker { m_directory->bitvectorLock() };
         m_directory->setIsUnswept(this, false);
-        m_directory->setIsDestructible(this, false);
+        m_directory->setIsDestructible(this, destructionMode != BlockHasNoDestructors && !isEmpty && m_directory->isDestructible(this));
```

Comment:
I see. I need to clear the bit once the block is cleared or block is refilled with objects w/o destructors. Nice!